go/consensus/tendermint/full/full.go

```diff
@@ -177,6 +177,8 @@ type fullService struct { // nolint: maligned
 	serviceClients   []api.ServiceClient
 	serviceClientsWg sync.WaitGroup
 
+	trackedRuntimes sync.Map
```

Comment:
Any reason why you are using a `sync.Map` instead of a regular map and a lock? Would make it more type safe.